src/capsule/CapsuleSigner.ts

```diff
@@ -189,29 +174,46 @@ export abstract class CapsuleBaseSigner implements Signer {
     return fromRpcSig(signatureHex)
   }
 
-  getNativeKey = () => this.account
+  public getNativeKey(): string {
+    if (!this.account) {
+      throw new Error('Native key not set')
+    }
+    return this.account
+  }
 
-  async decrypt(ciphertext: Buffer): Promise<Buffer> {
+  public async decrypt(ciphertext: Buffer): Promise<Buffer> {
     // TODO
     return Buffer.from('', 'base64')
   }
 
-  async computeSharedSecret(publicKey: string): Promise<Buffer> {
+  public async computeSharedSecret(publicKey: string): Promise<Buffer> {
     // TODO
     return Buffer.from('', 'base64')
   }
 
-  hexToBase64(hex: string) {
-    return Buffer.from(hex.replace('0x', ''), 'hex').toString('base64')
-  }
+  // --------------------------
 
-  base64ToHex(base64: string) {
-    return ensureLeading0x(Buffer.from(base64, 'base64').toString('hex'))
+  private async getWallet(userId: string, address: string): Promise<any> {
+    const response = await requestAndReauthenticate(
+      () => userManagementClient.getWallets(userId),
+      this.ensureSessionActive
+    )
+    for (const wallet of response.data.wallets) {
+      if (wallet.address && wallet.address.toLowerCase() == address.toLowerCase()) {
+        return wallet.id
+      }
+    }
+    return undefined
   }
-}
 
-export class CapsuleReactNativeSigner extends CapsuleBaseSigner {
-  protected getPrivateKeyStorage(account: string): PrivateKeyStorage {
-    return new PrivateKeyStorageReactNative(account)
+  private async prepSignMessage(userId: string, walletId: string, tx: string): Promise<any> {
```

Comment:
We should probably rename this to `preSignMessage` to be consistent with the client (though technically either could work)